Portrait clothing adjustments

VespertinePetrichor opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Removed unused `no_headgear_gfx`; was replaced in function by `prefer_no_lowborn_headgear`.
- Dragons no longer use generic headgear that clips through their ears (Crowns still worn)
- Unlinked gnomish goggles trigger from tenet_progressivism
  - It was causing all Mystery of the Makers court chaplains (has tenet_progressivism) to use goggles. 
    - Ergo, hyldnir and frostborn would also wear them.
  - Eligibility determined by heritage_goblin, heritage_gnome and creature_mechagnome.
- Assigned `female_cloaks_secular_fp1_nobility_01` to characters holding d_wardens or beholden to d_wardens.
  - [ ] **Optional** Needs recolour.
- Assigned `male_cloaks_secular_fp1_nobility_01` to Malfurion in new template `fp1_mystics_cloak`
  - `fp1_mystics_cloak` may be used for other characters, such as Medivh. However, it only fits him once he becomes a Prophet.
- Malfurion is now topless, as is the case in most of his official depictions.
  - **Known Issue** Malfurion's beard clips through the cloak.

## To-do list:
- [x]  Make night elves use purple clothing:
- [x]  Create the appropriate genes.
- [x]  Create the appropriate `.asset`.
- [x]  Create `night_elf` texture variations.
- [x]  Create color variations for the wardens cloaks.
- [ ]  Fix hair issues.
<!--
Before the merge, we recommend you to do these tests with your branch.
-->
## Tests:
- [x] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [x] The mod takes less than 5.5 GB in the Task Manager (Windows)

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
Check in-game (CK3 `1.5.0.2`) for the described visual gfx changes listed above.

VespertinePetrichor committed:
Deprecated clothing triggers

- `no_headgear_gfx` unused; previously replaced by `prefer_no_lowborn_headgear`
- Zaetaric/Cenaric now using WoW models

VespertinePetrichor committed:
Lowborn Dragon headgear

+ Dragon Ear Clipping

VespertinePetrichor committed:
Goblin goggles (Sans Mystery of Makers)

+ Goblins can also wear gnomish goggles now
- Mystery of the Makers no longer wears goggles; was inappropriate for Hyldnir, Frostborn and other cultures which may pick up tenet_progressivism

VespertinePetrichor committed:
Cloaks for Wardens

VespertinePetrichor committed:
Malfurion: Topless and in a Cloak

ercarp reviewed (approved):
Malfurion looks great now, I never even realized that he was supposed to be topless like his brother. I also love that you gave him a fur cloak, reminds me of his look in this artwork:

<details><summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/78575425/230559406-12581584-a086-4a89-a902-536c8d27b90e.png)

</details>

For scripted appearances, I usually just add them through `gfx\portraits\portrait_modifiers\wc_scripted_appearances` using the templates that already exist in Vanilla CK3, but I assume there's a reason why you made a new template for Malfurion's cloak. It does make it easier to choose just the one specific cloak, but you can always use values to pick the right one if there's more than one option within the template.

The following code would accomplish the same without creating a new clothing template:

<details><summary>Click to expand</summary>

```
malfurion_cloak = {
		dna_modifiers = {
			accessory = {
				mode = add
				gene = cloaks
				template = fp1_cloak
				value = 0.1
			}
		}
		weight = {
			base = 0
			modifier = {
				add = 100
				is_malfurion_trigger = yes
			}
		}
	}
```

</details>

But as you said, this may be reused for other characters so perhaps a new template is the way to go. 👍

I like the `d_wardens` cloak too; wish it was fur-lined like Maiev's actual cloak but sadly that doesn't exist in the game. To make it green, you need to make an `accessory_variation` - you can look in `gfx\portraits\accessory_variations` to see how this was done with the **Scourge** (`western`) and **Night Elf** (`byzantine`) clothing. However, I think they changed something about the way these work in `1.8` because Western clothes have graphical issues in `1.8`, so it might not be the best idea to do that in `1.5.0.2`.

> It was causing all Mystery of the Makers court chaplains (has tenet_progressivism) to use goggles. 

I remember seeing some Frost Vrykul using the Gnome Goggles a while back and thought it was a bit odd, so it's good to know this fixes that!

My own template rambling aside (which you can basically disregard as you more than likely know what you're doing), I don't see any problems here so I'll give it my approval. LGTM!

zumbak04 commented on common/culture/cultures/wc_azsharic.txt:
```diff
@@ -18,7 +18,7 @@
 	
 	coa_gfx = { western_coa_gfx wc_pagan_coa_gfx }
 	building_gfx = { mediterranean_building_gfx }
-	clothing_gfx = { creature_naga_gfx high_elven_clothing_gfx high_elven_hairstyles_gfx no_headgear_gfx }
```
I belive `no_headgear_gfx` wasn't removed, it was replaced by `no_lowborn_headgear_gfx`.
The logic behind ogres, elves, demons, draenei prefering no headgear is ears, horns cliping with headgears.

VespertinePetrichor replied:
I added `has_trait_with_flag = dragon_family` to the following trigger. Do you prefer I use `has_clothing_gfx = no_lowborn_headgear_gfx` instead? It would not be consistent with the other exceptions listed below.
```
prefer_no_lowborn_headgear = {
	OR = {
		# Ears get in the way
		culture = { has_clothing_gfx = no_lowborn_headgear_gfx }
		has_trait_with_flag = elven_family
		has_trait_with_flag = dragon_family
		is_race_no_gene_trigger = { RACE = creature_troll }
		is_race_no_gene_trigger = { RACE = creature_goblin }
		is_race_no_gene_trigger = { RACE = creature_ogre }
		is_race_no_gene_trigger = { RACE = creature_draenei }
		is_race_no_gene_trigger = { RACE = creature_sayaadi }
	}
}
```

zumbak04 commented on common/scripted_triggers/wc_clothing_triggers.txt:
```diff
@@ -295,7 +295,15 @@ religious_death_clothing_trigger = {
 religious_gnomish_clothing_trigger = {
-	faith = { has_doctrine_parameter = religious_gnomish_clothing }
+	OR = {
+		culture = {
+			OR = {
+				has_cultural_pillar = heritage_goblin
+				has_cultural_pillar = heritage_gnomish
+			}
+		}
```
Working with graphical look, use gfx triggers:

![image](https://user-images.githubusercontent.com/46384992/230788230-1624107d-6b18-429c-901b-e8c3798af4c0.png)

You assign gfx to culture, place gfx trigger in the code.

zumbak04 commented on gfx/portraits/portrait_modifiers/01_cloaks.txt:
```diff
@@ -156,6 +156,50 @@ cloaks = {
+			modifier = {
+				add = 100
+				portrait_fp1_warden_clothing_trigger = yes
+				culture = { has_cultural_pillar = heritage_darnassian }
```
I belive you don't need this trigger, the culture limit is handled in `portrait_fp1_warden_clothing_contents_trigger`.

VespertinePetrichor committed:
Merge branch 'master' into portrait-clothing-adjustments

VespertinePetrichor committed:
Reworked Triggers

+ Added clothing_gfx loc

Chomski2 committed:
Night elven colors

Night elven now use a custom color variation.
Warden cloaks now use a custom color variation.
Move the female high elven cloak to follow folder structure.

Chomski2 commented:
Tried to fix the errors from the log about the hair and beards but the majority are issues with the elven ear blendshapes of the hair and the new format paradox use for hairs.
I think it would be better to address this issues in a new PR since it would be necessary recreate the blendshapes.

helloruhe commented:
Closing due to inactivity. Open again and add to a project once this is being worked on again.
